include/pops/actions.hpp

```diff
@@ -0,0 +1,517 @@
+                    }
+                }
+            }
+            if (soil_pool_) {
+                // Get dispersers from the soil if there are any.
+                auto num_dispersers = soil_pool_->dispersers_from(i, j, generator);
+                // Put each disperser to the host pool.
+                for (int k = 0; k < num_dispersers; k++) {
+                    host_pool.disperser_to(i, j, generator.establishment());
+                }
+            }
+        }
+    }
+
+    /**
+     * @brief Activate storage of dispersers in soil
+     *
+     * Calling this function activates the soils. By default, the soil pool is not used.
+     * The parameters are soil pool used to store the dispersers and
+     * a percentage (0-1 ratio) of dispersers which will be send to the soil (and may
+     * establish or not depending on the soil pool object).
+     *
+     * Soil pool is optional and nothing is done when it is not set.
+     * This function needs to be called separately some time after the object is created
+     * to active the soil part of the simulation. This avoids the need for many
+     * more constructors or for many optional parameters which need default values.
+     *
+     * @param soil_pool Soils pool object to use for storage
+     * @param dispersers_percentage Percentage of dispersers moving to the soil
+     */
+    void activate_soils(
+        std::shared_ptr<SoilPool<IntegerRaster, FloatRaster, RasterIndex, Generator>>
+            soil_pool,
+        double dispersers_percentage)
+    {
+        this->soil_pool_ = soil_pool;
+        this->to_soil_percentage_ = dispersers_percentage;
+    }
+
+private:
+    /**
+     * Dispersal kernel
+     */
+    DispersalKernel& dispersal_kernel_;
+    /**
+     * Optional soil pool
+     */
+    std::shared_ptr<SoilPool<IntegerRaster, FloatRaster, RasterIndex, Generator>>
+        soil_pool_{nullptr};
+    /**
+     * Percentage (0-1 ratio) of disperers to be send to soil
+     */
+    double to_soil_percentage_{0};
+};
+
+/**
+ * Pest or pathogen individuals survive only with a given survival rate
+ *
+ * This removes a given percentage of exposed and infected in host pool.
+ */
+template<typename Hosts, typename IntegerRaster, typename FloatRaster>
+class SurvivalRateAction
+{
+public:
+    /**
+     * @brief Create object with a given survival rate.
+     *
+     * Survival rate is a spatially variable float raster with values between 0 and 1.
+     * From pest perspective this is the ratio of surviving individuals on hosts,
+     * while from the host perspective this is the ratio of hosts to keep infected or
+     * exposed.
+     *
+     * @param survival_rate Survival rate
+     */
+    SurvivalRateAction(const FloatRaster& survival_rate) : survival_rate_(survival_rate)
+    {}
+    /**
+     * Reduce the infection based on the pest survival rate.
+     *
+     * Infected and total number of exposed hosts are removed directly, while mortality
```

Comment:
Why mortality?

Reply:
Mortality cohorts are break down of total of infected just like the total exposed and exposed cohorts. I tried to clarify that in the new comment.